src/server/locals.js

```diff
@@ -0,0 +1,161 @@
+  this.session = new inspector.Session();
+  this.session.connect();
+
+  this.session.on('Debugger.paused', ({ params }) => {
+    if (params.reason == 'promiseRejection' || params.reason == 'exception') {
+      var key = params.data.description;
+      this.currentErrors.set(key, params);
+
+      if (this.currentErrors.size > 4) {
+        var firstKey = this.currentErrors.keys()[0];
+        this.currentErrors.delete(firstKey);
+      }
+    }
+  });
+
+  this.session.post('Debugger.enable', (_err, _result) => {
+    this.session.post('Debugger.setPauseOnExceptions', { state: 'all'}, (_err, _result) => {
+    });
+  });
+}
+
+Locals.prototype.currentLocalsMap = function() {
+  if (this.currentErrors.size) {
+    return new Map(this.currentErrors);
+  }
+}
+
+Locals.prototype.mergeLocals = function(localsMap, stack, key, callback) {
+  var matchedFrames;
+
+  try {
+    var localParams = localsMap.get(key);
+    matchedFrames = this.matchFrames(localParams, stack.reverse());
+  } catch (e) {
+    console.log(e);
+    return callback(e);
+  }
+
+  this.getLocalScopesForFrames(matchedFrames, function(err) {
+    callback(err);
+  });
+}
+
+// Finds frames in localParams that match file and line locations in stack.
+Locals.prototype.matchFrames = function(localParams, stack) {
+  var matchedFrames = [];
+  var localIndex = 0, stackIndex = 0;
+  var stackLength = stack.length;
+  var callFrames = localParams.callFrames;
+  var callFramesLength = callFrames.length;
+
+  for (; stackIndex < stackLength; stackIndex++) {
+    while (localIndex < callFramesLength) {
+      if (this.firstFrame(localIndex, stackIndex) || this.matchedFrame(callFrames[localIndex], stack[stackIndex])) {
+        matchedFrames.push({
+          stackLocation: stack[stackIndex],
+          callFrame: callFrames[localIndex]
+        });
+        localIndex++;
+        break;
+      } else {
+        localIndex++;
+      }
+    }
+  }
+
+  return matchedFrames;
+}
+
+Locals.prototype.firstFrame = function(localIndex, stackIndex) {
+  return !localIndex && !stackIndex;
+}
+
+Locals.prototype.matchedFrame = function(callFrame, stackLocation) {
+  if (!callFrame || !stackLocation) {
+    return false;
+  }
+
+  var position = stackLocation.runtimePosition;
+
+  // Node.js prefixes filename some URLs with 'file:///' in Debugger.callFrame,
+  // but with only '/' in the error.stack string. Remove the prefix to facilitate a match.
+  var callFrameUrl = callFrame.url.replace(/file:\/\//,'');
+
+  // lineNumber is zero indexed, so offset it.
+  var callFrameLine = callFrame.location.lineNumber + 1;
+  var callFrameColumn = callFrame.location.columnNumber;
+
+  return callFrameUrl === position.source &&
+    callFrameLine === position.line &&
+    callFrameColumn === position.column;
+}
+
+Locals.prototype.getLocalScopesForFrames = function(matchedFrames, callback) {
+  async.each(matchedFrames, this.getLocalScopeForFrame.bind(this), function (err) {
```

Comment:
seems that you can pass `callback` as the thrid arg to `async.each` rather than having the wrapping function - is there a reason for the extra wrapping function?

seems it will be called no matter the outcome the way it's written